cmd/cel-shed/p2p.go

```diff
@@ -75,3 +81,99 @@ var p2pPeerIDCmd = &cobra.Command{
 	},
 	Args: cobra.ExactArgs(1),
 }
+
+var (
+	errorOnAnyFailure bool
+	errorOnAllFailure bool
+)
+
+var p2pConnectBootstrappersCmd = &cobra.Command{
+	Use:   "connect-bootstrappers [network]",
+	Short: "Connect to bootstrappers of a certain network",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		network := p2p.GetNetwork(args[0])
+		bootstrappers, err := p2p.BootstrappersFor(network)
+		if err != nil {
+			return fmt.Errorf("failed to get bootstrappers: %w", err)
+		}
+
+		store := nodebuilder.NewMemStore()
+		cfg := p2p.DefaultConfig(node.Light)
+		modp2p := p2p.ConstructModule(node.Light, &cfg)
+
+		var mod p2p.Module
+		app := fx.New(
+			fx.NopLogger,
+			modp2p,
+			fx.Provide(fraud.Unmarshaler),
+			fx.Provide(cmd.Context),
+			fx.Provide(store.Keystore),
+			fx.Provide(store.Datastore),
+			fx.Supply(bootstrappers),
+			fx.Supply(network),
+			fx.Supply(node.Light),
+			fx.Invoke(func(modprov p2p.Module) {
+				mod = modprov
+			}),
+		)
+
+		if err := app.Start(cmd.Context()); err != nil {
+			return fmt.Errorf("failed to start app: %w", err)
+		}
+		defer func() {
+			if err := app.Stop(cmd.Context()); err != nil {
+				fmt.Printf("failed to stop application: %v\n", err)
+			}
+		}()
+
+		p2pInfo, err := mod.Info(cmd.Context())
+		if err != nil {
+			return fmt.Errorf("failed to get p2p info: %w", err)
+		}
+
+		fmt.Printf("PeerID: %s\n", p2pInfo.ID)
+		for _, addr := range p2pInfo.Addrs {
+			fmt.Printf("Listening on: %s\n", addr.String())
+		}
+		fmt.Println()
+
+		successfulConnections := 0
+		failedConnections := 0
+		for _, bootstrapper := range bootstrappers {
+			fmt.Printf("trying to connect to bootstrapper: %s\n", bootstrapper)
+			if err := mod.Connect(cmd.Context(), bootstrapper); err != nil {
+				fmt.Printf("failed to connect to bootstrapper: %v\n", err)
+				failedConnections++
+				continue
+			}
+			fmt.Println("connected")
+			successfulConnections++
+		}
+
+		if failedConnections == len(bootstrappers) && errorOnAllFailure {
+			fmt.Println()
+			fmt.Printf("failed to connect to all bootstrappers\n")
+			os.Exit(1)
+			return nil
+		} else if failedConnections > 0 && errorOnAnyFailure {
+			fmt.Println()
+			fmt.Printf("failed to connect to some bootstrappers\n")
+			os.Exit(1)
+			return nil
+		}
+
+		return nil
+	},
+	Args: cobra.ExactArgs(1),
+}
+
+func init() {
+	p2pConnectBootstrappersCmd.Flags().BoolVar(
+		&errorOnAnyFailure, "err-any", false,
+		"Return error if at least one bootstrapper is not reachable",
+	)
+	p2pConnectBootstrappersCmd.Flags().BoolVar(
+		&errorOnAllFailure, "err-all", false,
+		"Return error if no bootstrapper is reachable",
+	)
+}
```

Comment:
You may want to provide explicit timeout for the util either per connection or per bootsrapper. Otherwise it can potentially get stuck on connection attempt.